scripts/linters/pylint_extensions.py

```diff
@@ -2139,6 +2141,64 @@ def visit_classdef(self, node):
             self.add_message('no-dunder-metaclass', node=node)
 
 
+class StringConcatenationChecker(checkers.BaseChecker):
+    """Custom pylint checker which prohibits use of string concatenation and
+     promotes string interpolation instead.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'no-string-concatenation'
+    priority = -1
+    msgs = {
+        'C0035': (
+            'Please replace string concatenation with string interpolation.',
+            'string-interpolation',
+            'Enforce use of string interpolation over string concatenation.',
+        ),
+        'C0036': (
+            'Please remove the concatenation as it is not needed',
+            'no-string-concatenation',
+            'Enforce removal of unnecessary string concatenation.',
+        ),
+    }
+
+    def is_node_string_constant(self, node):
+        """Helper function for visit_binop. Called to check if any of the
+        operand is a string constant.
+        """
+        if (isinstance(node, astroid.nodes.Const) and
+                ('str' in node.pytype() or
+                 'unicode' in node.pytype())):
+            return True
+        return False
+
+    def visit_binop(self, node):
+        """Called for every '+' operator to prohibit usage of string
+         concatenation. It covers only those cases where at least one of the
+         operands is astroid.nodes.Const.
+
+        Args:
+            node: astroid.node.BinOp. Node to access module content.
+        """
+        if node.op != b'+':
+            return
+
+        is_left_node_a_str_const = self.is_node_string_constant(node.left)
+        is_right_node_a_str_const = self.is_node_string_constant(node.right)
+
+        if is_left_node_a_str_const and is_right_node_a_str_const:
+            self.add_message(
+                'no-string-concatenation',
+                node=node)
+        elif not (is_left_node_a_str_const or is_right_node_a_str_const):
+            return
+        else:
+            self.add_message(
+                'string-interpolation',
+                node=node)
```

Comment:
```suggestion
        if not (is_left_node_a_str_const or is_right_node_a_str_const):
            return
        
        if is_left_node_a_str_const and is_right_node_a_str_const:
            self.add_message('no-string-concatenation', node=node)
        else:
            self.add_message('string-interpolation', node=node)
```

Following the two principles:
1. End the function asap.
2. The first `if` case is going to be most of the situation in the codebase.

Reply:
@tanishq67  Is this comment resolved? (In general, make sure to reply to all the threads as suggested in the [wiki guidelines](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#address-review-comments-until-all-reviewers-give-lgtm-looks-good-to-me).

Reply:
I have taken your suggestions thanks! 
Sorry I thought by clicking on the button commit changes, github automatically resolves it. My bad!